scripts/render/dom-reducers.js

```diff
@@ -0,0 +1,150 @@
+    const gridItems = filterNodes(node.children, 'raqn-grid-item');
+
+    grids.map((grid, i) => {
+      const inicial = node.children.indexOf(grid);
+      const nextGridIndex = grids[i + 1] ? node.children.indexOf(grids[i + 1]) : node.children.length;
+      gridItems.map((item) => {
+        const itemIndex = node.children.indexOf(item);
+        // get elements between grid and item and insert into grid
+        if (itemIndex > inicial && itemIndex < nextGridIndex) {
+          const children = node.children.splice(inicial + 1, itemIndex - inicial);
+          const gridItem = children.pop(); // remove grid item from children
+          gridItem.children = children;
+          grid.children.push(gridItem);
+        }
+      });
+      return grid;
+    });
+  }
+  return node;
+};
+// Compare this snippet from scripts/render/dom.js:
+
+export const recursive = (fn) => (nodes, level) =>
+  nodes.map((node) => {
+    if (node.children) {
+      node.children = recursive(fn)(node.children, level + 1);
+    }
+    return fn(node, level);
+  });
+
+// eslint-disable-next-line prefer-destructuring
+export const toWebComponent = (node) => {
+  Object.keys(componentList).forEach((componentClass) => {
+    if ((node.class && node.class.includes(componentClass)) || node.tag === componentClass) {
+      const { dependencies } = componentList[componentClass];
+      if (componentList[componentClass].transform) {
+        // eslint-disable-next-line no-param-reassign
+        node = componentList[componentClass].transform(node);
+      } else {
+        node.tag = componentList[componentClass].tag;
+      }
+
+      if (!loadedComponents[componentClass]) {
+        loadedComponents[componentClass] = componentList[componentClass];
+      }
+      if (dependencies) {
+        dependencies.forEach((dependency) => {
+          if (!loadedComponents[dependency]) {
+            loadedComponents[dependency] = componentList[dependency];
+          }
+        });
+      }
+    }
+  });
+  return node;
+};
+
+// load modules in order of priority
+
+export const loadModules = (nodes, modules = loadedComponents) => {
+  window.inicialization = Object.keys(modules)
+    .sort((a, b) => modules[a].priority - modules[b].priority)
+    .map((component) => {
+      const { script, tag, priority } = modules[component];
+      if (window.raqnComponents[tag]) return window.raqnComponents[tag].default;
+      return new Promise((resolve) => {
+        setTimeout(async () => {
+          const { js, css } = loadModule(script);
+
+          const mod = await js;
+          const style = await css;
+          if (mod.default.prototype instanceof HTMLElement) {
+            if (!window.customElements.get(tag)) {
+              window.customElements.define(tag, mod.default);
+              window.raqnComponents[tag] = mod.default;
+            }
+          }
+          resolve({ tag, mod, style });
+        }, 10 + priority);
```

Comment:
should we really wait 10+ms for loading?

Reply:
No, was due to avoiding negative values but it's dine without 10